packages/test/src/test-integration-update-validation-interceptors.ts

```diff
@@ -0,0 +1,74 @@
+import * as wf from '@temporalio/workflow';
+import { Next, UpdateInput, WorkflowInboundCallsInterceptor } from '@temporalio/workflow';
+import { helpers, makeTestFunction } from './helpers-integration';
+
+const update = wf.defineUpdate<string[], [string]>('update');
+const doneUpdate = wf.defineUpdate<void, []>('done-update');
+
+export async function workflowWithUpdates(): Promise<string[]> {
+  const state: string[] = [];
+  const updateHandler = async (arg: string): Promise<string[]> => {
+    state.push(arg);
+    if (arg === 'fail-update') {
+      throw new wf.ApplicationFailure(`Deliberate ApplicationFailure in handler`);
+    }
+    return state;
+  };
+  const doneUpdateHandler = (): void => {
+    state.push('done');
+  };
+  const validator = (arg: string): void => {
+    if (arg === 'bad-arg') {
+      throw new Error('Validation failed');
+    }
+  };
+  wf.setHandler(update, updateHandler, { validator });
+  wf.setHandler(doneUpdate, doneUpdateHandler);
+  await wf.condition(() => state.includes('done'));
+  state.push('$');
+  return state;
+}
+
+const test = makeTestFunction({
```

Comment:
Move `const test = ...` very early in this file, and move workflow function near the tests that depends on it (there could be mulitple workflow functions per test file)

Reply:
Thanks, I'm not sure how the `makeTestFunction` got down there. I think I prefer to keep the workflow definition before the interceptors, since it's the latter that are the focus of the test and the reader needs to understand the role they play.